### PR TITLE
Make af optional

### PR DESCRIPTION
**Pull request by RaphaelKreft**
Implementation to make Anglefield optional in our pipeline.

### CODE REVIEW

File: gluefactory/models/extractors/joint_point_line_extractor.py [hunk 1]
```diff
@@ -159,21 +159,23 @@ def _init(self, conf):
...
-                kernel_size=3,
-                padding=1,
-            ),
-            nn.ReLU(),
-            nn.BatchNorm2d(conf.line_af_decoder_channels),
-            nn.Conv2d(conf.line_af_decoder_channels, 1, kernel_size=1),
-            nn.Sigmoid(),
-        )
+        # only use line angle-field if configured
+        if self.conf.use_line_anglefield:
```
[the remark this answers is not in the document]
Comment (RaphaelKreft) [reply gluefactory/models/extractors/joint_point_line_extractor.py:163]: Fair, will add :)

File: gluefactory/models/extractors/joint_point_line_extractor.py [hunk 2]
```diff
@@ -429,12 +432,15 @@ def _forward(self, data):
...
                     keypoints_deeplsd = torch.cat((deeplsd_lines_torch[:,0],deeplsd_lines_torch[:,1]))
+                # prepare line data for line detection!
                 line_data = {
                      "points": torch.clone(kp) if not self.conf.line_detection.use_deeplsd_kp else keypoints_deeplsd,
                      "distance_map": torch.clone(df) if not self.conf.line_detection.use_deeplsd_df_af else deeplsd_output["df"][0],
-                     "angle_map": torch.clone(af) if not self.conf.line_detection.use_deeplsd_df_af else deeplsd_output["line_level"][0],
                      "descriptors": torch.clone(desc) if not self.conf.line_detection.use_deeplsd_kp else torch.zeros((keypoints_deeplsd.shape[0],128)).cuda(),
                 }
+                if self.conf.use_line_anglefield:
+                    line_data["angle_map"] = torch.clone(af) if not self.conf.line_detection.use_deeplsd_df_af else deeplsd_output["line_level"][0]
```
Comment (hardik01shah) [inline gluefactory/models/extractors/joint_point_line_extractor.py:442]: Pass None if AF is not computed

**Review by hardik01shah — CHANGES_REQUESTED** [review]
One important thing: If AF is not computed, pass None / zeros to pold2_extractor. Otherwise good!

**Comment by RaphaelKreft** [comment]
I have merged tge branch BruteForcing-Approach-LineDetection branch into this one to have an up to date line detection
@hardik01shah @FelixMoeller3 In suggest we review current state of this brnahc here as well and merge it. Hardik can then work on line detection further.

File: gluefactory/configs/eval/jpldd+BF.yaml
Comment (RaphaelKreft) [inline gluefactory/configs/eval/jpldd+BF.yaml:1]: @hardik01shah Shouldnt this be named jpldd+LM? or what is the BF standing for?

Comment (hardik01shah) [reply gluefactory/configs/eval/jpldd+BF.yaml:1]: BF stands for the BruteForce approach

**Review by hardik01shah — APPROVED** [review]
LGTM

**Pull request merged**